From 709a2bbce44d04e9e81f0ac94e4df64e4a43bc54 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 12 Sep 2022 17:47:42 +0200
Subject: [PATCH 0332/1695] dts: stm32: populate the STGEN node in
 stm32mp251.dtsi

Populate the STGEN node in the stm32mp251.dtsi SOC device tree source file.
This node mandates the "clocks" property with the three STGEN associated
clocks.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 149bb171bfa5f41a674102e04c7080c73e06e20c dts: stm32: populate the STGEN node in stm32mp251.dtsi]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I683abfeab3608fa068f9dbacdc65f21084e84a30
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/265964
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 95e938c27..907f6285f 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -554,9 +554,12 @@
 			status = "disabled";
 		};
 
-		stgen: stgen@48080000 {
+		stgenc: stgen@48080000 {
+			compatible = "st,stm32mp25-stgen";
 			reg = <0x48080000 0x1000>;
-			status = "disabled";
+			clocks = <&rcc CK_BUS_STGEN>, <&rcc CK_KER_STGEN>;
+			clock-names = "bus", "tsgen_clk";
+			st,tsgen-clk-source = <&rcc HSE_CK>;
 		};
 
 		fmc: memory-controller@48200000 {
-- 
2.47.2

